sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -342,6 +384,103 @@ pi_result piDeviceGetInfo(pi_device device, pi_device_info paramName,
     return PI_SUCCESS;
   }
 
+  case PI_DEVICE_INFO_PARTITION_PROPERTIES: {
+    // SYCL spec says: if this SYCL device cannot be partitioned into at least
+    // two sub devices then the returned vector must be empty.
+    pi_uint32 partitionMaxSubDevices = 0;
+    if (device->subLevel == -1)
+      device->subLevel = getSubLevel(device);
+    if (device->isRootDevice()) {
+      clGetDeviceInfo(
+          cast<cl_device_id>(device), CL_DEVICE_PARTITION_MAX_SUB_DEVICES,
+          sizeof(partitionMaxSubDevices), &partitionMaxSubDevices, nullptr);
+    } else if (device->isSubDevice()) {
+      // find out number of CCSes
```

Comment:
What if it's already a sub-sub-device ?

Reply:
Then it will just return 0 in out_num_devices

Reply:
I think I'd prefer to be more explicit about this. Something like

```
switch (level) {
  case 2: return no_further_partitioning
  case 0: return delegate_to_opencl
  case 1: { our logic }
  default:
    assert/PI_INVALID_ERROR/etc.
}
```

Reply:
Modified in upcoming patch. Thanks